fe/fe-core/src/main/java/com/starrocks/statistic/StatisticAutoCollector.java

```diff
@@ -69,101 +55,139 @@ protected void runAfterCatalogReady() {
 
         initDefaultJob();
 
-        List<TableCollectJob> allJobs = generateAllJobs();
+        updateHealthy();
+
+        List<TableCollectJob> allJobs;
+        if (Config.enable_collect_full_statistics) {
+            allJobs = generateFullAnalyzeJobs();
+        } else {
+            allJobs = generateAllJobs();
+        }
 
-        collectStatistics(allJobs);
+        for (TableCollectJob tcj : allJobs) {
+            statisticExecutor.collectStatistics(tcj);
+        }
 
         expireStatistic();
     }
 
     private void initDefaultJob() {
-        // Add a default sample job if was't been collect
+        // Add a default sample job if wasn't collect
         List<AnalyzeJob> allAnalyzeJobs = GlobalStateMgr.getCurrentAnalyzeMgr().getAllAnalyzeJobList();
         if (allAnalyzeJobs.stream().anyMatch(j -> j.getScheduleType() == ScheduleType.SCHEDULE)) {
             return;
         }
 
-        AnalyzeJob analyzeJob = new AnalyzeJob();
-        // all databases
-        analyzeJob.setDbId(AnalyzeJob.DEFAULT_ALL_ID);
-        analyzeJob.setTableId(AnalyzeJob.DEFAULT_ALL_ID);
-        analyzeJob.setColumns(Collections.emptyList());
-        analyzeJob.setScheduleType(ScheduleType.SCHEDULE);
-        analyzeJob.setType(AnalyzeType.SAMPLE);
-        analyzeJob.setStatus(ScheduleStatus.PENDING);
-        analyzeJob.setWorkTime(LocalDateTime.MIN);
-
+        AnalyzeJob analyzeJob = new AnalyzeJob(AnalyzeJob.DEFAULT_ALL_ID, AnalyzeJob.DEFAULT_ALL_ID, Collections.emptyList(),
+                AnalyzeType.SAMPLE, ScheduleType.SCHEDULE, Maps.newHashMap(), ScheduleStatus.PENDING, LocalDateTime.MIN);
         GlobalStateMgr.getCurrentAnalyzeMgr().addAnalyzeJob(analyzeJob);
     }
 
-    private void collectStatistics(List<TableCollectJob> allJobs) {
-        // AnalyzeJob-List<TableCollectJob> for update AnalyzeJob status
-        Map<AnalyzeJob, List<TableCollectJob>> analyzeJobMap = Maps.newHashMap();
+    private void updateHealthy() {
```

Comment:
Don't understand this logic

Reply:
The logic of this code has been written in the comments.
/*
  * health = totalLoadRows / totalRowCount.
  * The ratio of the number of modified lines to the total number of lines.
  * Because we cannot obtain complete table-level information, we use the row count of
  * the partition with the smallest row count as totalRowCount.
  * It can be understood that we assume an extreme case where all imported and modified lines
  * are concentrated in only one partition
  */